src/lxc/string_utils.c

```diff
@@ -519,6 +519,7 @@ int lxc_grow_array(void ***array, size_t *capacity, size_t new_size, size_t capa
 	/* first time around, catch some trivial mistakes of the user
 	 * only initializing one of these */
 	if (!*array || !*capacity) {
+		free(*array);
```

Comment:
That whole block confuses me. This makes it look like we expect the caller to have an uninitialized pointer and then initialize it? That whole block doesn't make much sense to me.
Wouldn't it be better if we'd report an error here instead of freeing the caller's memory behind their back?

Reply:
if we have for loop to call lxc_grow_array many times,we initialize it every time,the old don't free, I think it will memory leek